lib/Basics/DownCast.h

```diff
@@ -36,7 +36,9 @@ constexpr auto* downCast(From* from) noexcept {
   static_assert(!std::is_reference_v<To>, "'To' shouldn't be reference");
   using CastTo =
       std::conditional_t<std::is_const_v<From>, std::add_const_t<To>, To>;
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
```

Comment:
Why is this necessary? `TRI_ASSERT` is only active in maintainer mode anyway?

Reply:
It's my paranoia, now TRI_ASSERT always generate some code, yes it's always false, but there is not guarantee at compile time

Reply:
:-) Fine.